src/server/db_slice.cc

```diff
@@ -55,6 +55,13 @@ void PerformDeletion(PrimeIterator del_it, ExpireIterator exp_it, EngineShard* s
   if (pv.ObjType() == OBJ_STRING)
     stats.strval_memory_usage -= value_heap_size;
 
+  if (ClusterConfig::IsClusterEnabled()) {
+    string key;
+    del_it->first.GetString(&key);
```

Comment:
Since we only need the key temporarily, we could use `GetSlice()` here to potentially duck an allocation